File: C++/find_primes/find_primes.cpp
#include <iostream>
#include <vector>
#include <ctime>

bool is_prime(int num)
{
    if (num <= 1)
        return false;
    for (int i = 2; i * i <= num; ++i)
    {
        if (num % i == 0)
            return false;
    }
    return true;
}

int find_primes(int start, int end)
{
    int count = 0;
    for (int num = start; num <= end; ++num)
    {
        if (is_prime(num))
        {
            count++;
        }
    }
    return count;
}

int main()
{
    int start = 0;
    int end = 10000000;
    std::vector<int> primes;

    // Find primes and measure execution time
    clock_t start_time = clock();
    int primes_count = find_primes(start, end);
    clock_t end_time = clock();

    double execution_time = static_cast<double>(end_time - start_time) / CLOCKS_PER_SEC;
    std::cout << "Execution time: " << execution_time << " seconds" << std::endl;
    std::cout << "Total prime numbers found: " << primes.size() << std::endl;

    return 0;
}

File: Rust/find_primes/find_primes.rs
use std::time::Instant;

fn is_prime(num: i32) -> bool {
    if num <= 1 {
        return false;
    }
    for i in 2..=((num as f64).sqrt() as i32) {
        if num % i == 0 {
            return false;
        }
    }
    true
}

fn find_primes(start: i32, end: i32) -> i32 {
    let mut count = 0;
    for num in start..=end {
        if is_prime(num) {
            count += 1;
        }
    }
    count
}

fn main() {
    let start = 0;
    let end = 10000000;

    // Find primes and measure execution time
    let start_time = Instant::now();
    let primes_count = find_primes(start, end);
    let end_time = Instant::now();

    let execution_time = end_time.duration_since(start_time).as_secs_f64();
    println!("Execution time: {} seconds", execution_time);
    println!("Total prime numbers found: {}", primes_count);
}
